app/uk/gov/hmrc/gform/eval/EvaluationResults.scala

```diff
@@ -880,19 +884,23 @@ case class EvaluationResults(
     implicit val m = evaluationContext.messages
 
     def loop(expr: Expr): ExpressionResult = expr match {
-      case Add(field1: Expr, field2: Expr) => loop(field1) + loop(field2)
+      case Typed(expr: Expr, _)                    => loop(expr)
+      case Add(field1: Expr, field2: Expr)         => loop(field1) + loop(field2)
+      case Subtraction(field1: Expr, field2: Expr) => loop(field1) - loop(field2)
+      case Divide(field1: Expr, field2: Expr)      => loop(field1) / loop(field2)
+      case Multiply(field1: Expr, field2: Expr)    => loop(field1) * loop(field2)
```

Comment:
These are not valid operations on `Period` and this should not be here. See the other comment.